### ISSUE TITLE
Implement the Upload processing status UI component

### ISSUE THREAD

**Issue by Adal3n3** [codecov/engineering-team#2522]
Figma: https://www.figma.com/design/4Z7yb2dkIIATkfzpWoMYQq/GH-2220?node-id=1-2

![Image](https://github.com/user-attachments/assets/982b6845-f4e7-4c2d-9236-d95aecdc0c8b)

---

![Image](https://github.com/user-attachments/assets/2793e32d-8f8a-40ac-807d-7f4d2303ce1f)

**Comment by Adal3n3** [comment]
@thomasrockhu-codecov @trent-codecov can you assign this issue?

**Comment by Adal3n3** [comment]
I reviewed the upload processing status UI with @trent-codecov and @rohan-at-sentry, and Trent checked our database to confirm we have the data available to display in the UI so this issue might not need platform work. We will wait until @spalmurray-codecov works on this issue to see if he has all the resources he needs. No action needed for now.

**Comment by Adal3n3** [comment]
Synced with @spalmurray-codecov on 10/4:
- confirmed that we are good and ready to implement the upload processing status UI.
- The grey bar that indicating we are still waiting for uploads, we can use an estimate timestamp. 
- @Adal3n3 will open a new issue for the code snippet. We can display the code snippets as soon as we detect an error, so users don't have to wait for the entire processing to finish.
- Without the code snippets, just the processing status UI, takes 3-5 days eng work.
- For Code snippet - unknown but might take 3-5 days eng work.

**Comment by Adal3n3** [comment]
Hi @spalmurray-codecov, why this issue is blocked? Can you share the reason?

**Comment by spalmurray-codecov** [comment]
Hi @Adal3n3! This is blocked as I am figuring out a solution to the estimation problem. Here's the doc that I got some feedback on- https://www.notion.so/sentry/Estimating-the-expected-number-of-uploads-for-a-commit-1318b10e4b5d80228157e81fed87dd7f

I need to think a little more about the solution here before we can continue. 

That said, I think it makes sense to make this a Q4 OKR though. There's enough work here to warrant a sprint or two of work no matter what we decide for solution.

**Comment by Adal3n3** [comment]
Thank you @spalmurray-codecov for sharing the blocker. We only want this 'processing upload' component to display while uploads are in progress and uploads completed with error(s), but we don’t have a way to determine when uploads are complete. 
--> Next steps: I’ll set up a meeting to explore possible solutions with the team.

[_This is not a blocker_] Without the data for estimation of uploads, we can deliver a v1 solution with a static width for all statuses, including the gray bar. This means all statuses will display at 100% width and won’t visually represent "progress". In v2, we can explore using the previous upload count as an estimate for the gray bar.

V1 solution: without estimation of uploads
![Image](https://github.com/user-attachments/assets/ea4cb333-0824-42ae-a767-7e74c402c292)

(cc @codecovdesign )

**Comment by Adal3n3** [comment]
This issue has a dependency on https://github.com/codecov/engineering-team/issues/2702 to figure out the **start and end state** of processing uploads. We want to understand if it's possible to determine the start/end of processing uploads as an initial next step.

**Comment by spalmurray-codecov** [comment]
Blocked pending https://github.com/codecov/engineering-team/issues/2921

**Comment by Adal3n3** [comment]
@spalmurray-codecov We still can communicate to the user without showing the estimation of uploads. I believe the true blocker here is without knowing the start and end state of processing uploads, we don't know when to show the upload processing status UI.

**Comment by Adal3n3** [comment]
Follow up the discussion from here https://github.com/codecov/engineering-team/issues/2702#issuecomment-2542265121
We currently do not have distinct states for the start and end of upload processing, but we do have a `processing` state to indicate that an upload has started. The challenge is that we don’t know precisely when an upload is complete, so displaying a "processing" UI alongside the coverage report could be misleading. As an alternative, we propose communicating pending uploads by displaying **the number of uploads in progress**. (cc @spalmurray-codecov )

Figma: https://www.figma.com/design/4Z7yb2dkIIATkfzpWoMYQq/GH-2220?node-id=1-2

![Image](https://github.com/user-attachments/assets/982b6845-f4e7-4c2d-9236-d95aecdc0c8b)

---

![Image](https://github.com/user-attachments/assets/2793e32d-8f8a-40ac-807d-7f4d2303ce1f)

**Comment by rohan-at-sentry** [comment]
@Adal3n3 can you review if https://github.com/codecov/engineering-team/issues/3229 captures what the final design is going to look like? Closing this out.

**Comment by Adal3n3** [comment]
2/3 quick synced with @spalmurray-codecov @rohan-at-sentry for the gh issue migration. Sounds like we are all aligned.
